### PR TITLE
Created autopolling producer

### PR DESCRIPTION
**Pull request by Pchelolo**
Better encapsulation of `poll` call and better promise rejection handling. Also for testing decreased the side of the buffer - if poll is not called correctly tests would start failing.

cc @wikimedia/services

### CODE REVIEW

File: sys/kafka.js [hunk 1]
```diff
@@ -93,9 +93,8 @@ class Kafka {
 
             // TODO: This has to return a promise for guaranteed producer.
             // TODO: This is a quick fix to be rewritten.
-            this.producer.produce(`${this.kafkaFactory.produceDC}.${message.meta.topic}`, 0,
+            return this.producer.produce(`${this.kafkaFactory.produceDC}.${message.meta.topic}`, 0,
```
Comment (d00rman) [inline sys/kafka.js]: The above TODOs can now be removed, couldn't they?

File: lib/kafka_factory.js [hunk 2]
```diff
@@ -107,6 +107,34 @@ class GuaranteedProducer extends kafka.Producer {
...
+            try {
+                const result = super.produce(topic, partition, message, key);
+                if (result !== true) {
+                    return reject(result);
+                }
+                return resolve(result);
+            } catch (e) {
+                return reject(e);
+            } finally {
+                this.poll();
```
Comment (d00rman) [inline lib/kafka_factory.js]: Are we sure that this will be called before the returned Promise is being activated?

Comment (Pchelolo) [inline lib/kafka_factory.js]: Yep, just checked that.

File: sys/kafka.js (hunk 1 above)
Comment (Pchelolo) [inline sys/kafka.js]: Oh, sorry, forgot about it!

**Pull request merged**